src/components/HistoricalForecast/HistoricalForecast.tsx

```diff
@@ -1,6 +1,95 @@
-import DateRangePicker from 'components/DateRangePicker';
-import * as classes from './styles';
+import { useContext } from 'react';
+// import Button from '@mui/material/Button';
+import Grid from '@mui/material/Grid';
+import { Field, Formik, Form } from 'formik';
+import * as yup from 'yup';
+import SelectField from 'components/SelectField';
+import LANGUAGE_OPTIONS from 'constants/languageOptions';
+import Languages from 'constants/languages';
+import TEMPERATURE_UNITS_OPTIONS from 'constants/temperatureUnitsOptions';
+import TemperatureUnits from 'constants/temperatureUnits';
+import SettingsContext from 'contexts/SettingsContext';
+import Location from 'types/location';
+import SettingsContextType from 'types/settingsContextType';
+
+const LANGUAGE_CHOICE_LABEL_ID = 'language-label';
+const LANGUAGE_CHOICE_LABEL = 'Language';
+// const LOCATION_AUTOCOMPLETE = 'location-select';
+const TEMPERATURE_UNITS_LABEL_ID = 'unit-label';
+const TEMPERATURE_UNITS_LABEL = 'Temperature unit';
+
+interface Validation {
+  language: Languages;
+  startDate: Date;
+  endDate: Date;
+  location: Location;
+  temperatureUnit: TemperatureUnits;
+}
+
+const VALIDATION_SCHEMA: yup.ObjectSchema<Validation> = yup.object().shape({
+  language: yup.mixed<Languages>().required('Required'),
+  startDate: yup.date().required('Required'),
```

Comment:
Don't forget interdependent validation: `startDate` shouldn't be greater `endDate`. Also both dates should be in the past; for it create reusable handler